datahub-web-react/src/app/ingest/source/builder/RecipeForm/constants.ts

```diff
@@ -0,0 +1,312 @@
+    recipe: any,
+    value: string | boolean | null | undefined,
+    field: string,
+    parentField?: string,
+) {
+    if (value !== undefined) {
+        const recipeValue = value === null ? undefined : value; // clear out fields with null values
+        const updatedRecipe = { ...recipe };
+        if (!updatedRecipe.source.config) updatedRecipe.source.config = {};
+
+        if (recipeValue === undefined && parentField) {
+            // clearing the field and possibly its parent
+            const finalRecipe = updateParentAndField(updatedRecipe, field, parentField);
+            return finalRecipe;
+        }
+
+        if (parentField) {
+            if (!updatedRecipe.source.config[parentField]) updatedRecipe.source.config[parentField] = {};
+            updatedRecipe.source.config[parentField][field] = recipeValue;
+        } else {
+            updatedRecipe.source.config[field] = recipeValue;
+        }
+
+        return updatedRecipe;
+    }
+    return recipe;
+}
+
+export function setListValuesOnRecipe(recipe: any, values: string[], field: string, parentField: string) {
+    if (values !== undefined) {
+        const updatedRecipe = { ...recipe };
+        if (!updatedRecipe.source.config) updatedRecipe.source.config = {};
+
+        const filteredValues: string[] | undefined = values.filter((v) => !!v);
+        if (!filteredValues.length) {
+            // clearing the field and possibly its parent
+            const finalRecipe = updateParentAndField(updatedRecipe, field, parentField);
+            return finalRecipe;
+        }
+
+        if (!updatedRecipe.source.config) updatedRecipe.source.config = {};
+        if (!updatedRecipe.source.config[parentField]) updatedRecipe.source.config[parentField] = {};
+        updatedRecipe.source.config[parentField][field] = filteredValues;
+        return updatedRecipe;
+    }
+    return recipe;
+}
+
+export const ACCOUNT_ID: RecipeField = {
+    name: 'account_id',
+    label: 'Account ID',
+    tooltip: 'Snowflake account. e.g. abc48144',
+    type: FieldType.TEXT,
+    rules: null,
+    getValueFromRecipe: (recipe: any) => recipe.source.config?.account_id,
+    setValueOnRecipe: (recipe: any, value: string) => setFieldValueOnRecipe(recipe, value, 'account_id'),
+};
+
+export const WAREHOUSE: RecipeField = {
+    name: 'warehouse',
+    label: 'Warehouse',
+    tooltip: 'Snowflake warehouse.',
+    type: FieldType.TEXT,
+    rules: null,
+    getValueFromRecipe: (recipe: any) => recipe.source.config?.warehouse,
+    setValueOnRecipe: (recipe: any, value: string) => setFieldValueOnRecipe(recipe, value, 'warehouse'),
+};
+
+export const USERNAME: RecipeField = {
+    name: 'username',
+    label: 'Username',
+    tooltip: 'Snowflake username.',
+    type: FieldType.TEXT,
+    rules: null,
+    getValueFromRecipe: (recipe: any) => recipe.source.config?.username,
+    setValueOnRecipe: (recipe: any, value: string) => setFieldValueOnRecipe(recipe, value, 'username'),
```

Comment:
it feels like these two lines are redundant. can't we just have a `fieldPath` property that supplies the fieldpath that these two lambdas both reference?